### ISSUE TITLE
[Guido's Gorgeous Lasagna]: Unable to Complete the Final Task (adding comments) for Python 

### ISSUE THREAD

**Issue by Ollie-Judge** [exercism/python#3161]
i have 5 parts to the question and on the last part its asking me for adding comments, which i have done, i know the answers to the questions are right and ive checked the whole code area for pep8 and pep 20 compliance and cant seem to get the success message, so, i cant help but feel that this is a bug

**Comment by BethanyG** [comment]
Hi @Ollie-Judge 👋🏽 

Thanks for filing this issue.  If you don't mind, could you add in the code that you were testing & a screenshot of the test results you are getting in the UI?  

Although this exercise has not changed recently, there may indeed be an issue with the tests --  but I need to see your solution to test that hypothesis out /be sure it's a bug.

Many thanks in advance 🌟

**Comment by Ollie-Judge** [comment]
here are the screenshots that i took, the error.png file should show the code and the error-messages.png file should show the error messages

![error](https://user-images.githubusercontent.com/78495702/185344118-3d9ce155-a2d2-4e81-93b4-c3d5c5f1f7a0.png)
![error-messages](https://user-images.githubusercontent.com/78495702/185344120-28ef91cf-26f7-4c3a-9e2d-46caeee8ea0f.png)

**Comment by bobahop** [comment]
It would be helpful if the code for the failing task were pasted into a comment as text. More than a dozen solutions have been successfully published within the last 24 hours, so to be able to look at the code  and run it would be helpful. The screenshot of the code isn't the best suited for that. Thanks!

**Comment by bobahop** [comment]
Scratch that. I think the problem is a missing docstring that's expected.

**Comment by bobahop** [comment]
Some resources for how to write docstrings can be found [here](https://www.programiz.com/python-programming/docstrings) and [here](https://www.geeksforgeeks.org/python-docstrings/).

It's good to know [the distinctions between comments and docstrings](https://www.pythonforbeginners.com/comments/difference-between-comments-and-docstrings-in-python).

**Comment by bobahop** [comment]
Actually, it's not that the docstring is missing so much as it's located in the wrong place. Underneath the code instead of just below the function declaration.

**Comment by BethanyG** [comment]
@bobahop - A couple of things to note here:

1.  It's always best to link to the [Python Documentation](https://docs.python.org/3/index.html) or relevant [PEPs](https://peps.python.org/pep-0000/) when talking about standards. Both of the resources you've linked for writing docstrings are somewhat incorrect.  While triple single quotes will indeed work with the Python interpreter, [PEP257](https://peps.python.org/pep-0257/)  notes:
> For consistency, always use """triple double quotes""" around docstrings. Use r"""raw triple double quotes""" if you use any backslashes in your docstrings. For Unicode docstrings, use u"""Unicode triple-quoted strings""".

[PEP8](https://peps.python.org/pep-0008/#string-quotes) notes (_emphasis mine_):

>In Python, single-quoted strings and double-quoted strings are the same. This PEP does not make a recommendation for this. Pick a rule and stick to it. When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string. It improves readability.
>
>**For triple-quoted strings, always use double quote characters to be consistent with the docstring convention** in [PEP 257](https://peps.python.org/pep-0257).

I got caught by this earlier this year, and we had to go back and change many of our code stubs because they used single triple quotes which was a PITA. Writing docstrings with single triple quotes may also mess with automated documentation systems or testing/test runners. 

2.  Please try not to do multiple comments right after one another in a GitHub issue unless you really need to.  Comments send emails/notifications to everyone watching the repo or involved in the discussion, which can get quite "noisy" and hard to follow.  Thanks!

@Ollie-Judge  -

We were perhaps not as clear as we needed to be in the instructions for this exercise.  Python doesn't have multi-line comments.  Every comment line must start with `#`.  What  we were asking (and testing for) in this final task were **docstrings** - special strings that are interpreted as comments (and in some cases tests).

The format of your attempted docstrings looks good.  Unfortunately, they're in the wrong position to be picked up as docstring comments, and are being seen by the interpreter as plain multi-line `strings`.  In order for the Python interpreter to read multi-line strings as _**docstrings**_ and place them in the function attribute dictionary, they need to be positioned **_directly below the line that defines a function or class_**:

```python

def  elapsed_time_in_minutes(layers, elapsed_bake_time):
    """
    Return elapsed cooking time.

    This function takes two numbers representing the 
    number of layers & the time already spent
    baking and calculates the total elapsed minutes
    spent cooking the lasagna.
    """ 

    return preparation_time_in_minutes(layers) + elapsed_bake_time
```

Our tests for this exercise validate by calling ` function.__doc__`,  so if the function attribute dictionary doesn't have a value for the `__doc__` key, it throws an `unexpectedly None` error, which is what that `assertsisNotNone` is in the test: ` self.assertIsNotNone(function.__doc__, msg=failure_msg)`.   

But that's not at all clear to someone not familiar with  Python test code, so we probably need to both clarify the instructions and put comments in the test for students.  😄 

When I have time this weekend, I will look into clarifying the instructions and commenting the test - unless you'd like to try out making a PR?  If you would like to PR changes, I am happy to help you through the process - just let me know!

**Comment by BethanyG** [comment]
Leave my issue alone, stale bot!
